libkv.go

```diff
@@ -30,29 +33,98 @@ func NewLibKV(url *url.URL) *LibKV {
 
 	sourceType = ""
 	client = ""
+	options := store.Config{}
 
 	if url.Scheme == "consul" {
 		consul.Register()
 		sourceType = store.CONSUL
 		client = env.Getenv("CONSUL_HTTP_ADDR", "localhost:8500")
+		if timeout := env.Getenv("CONSUL_TIMEOUT", ""); timeout != "" {
+			num, err := strconv.ParseInt(timeout, 10, 16)
+			if err != nil {
+				logFatal("consul timeout error", err)
+			}
+			options.ConnectionTimeout = time.Duration(num) * time.Second
+		}
+		if ssl := env.Getenv("CONSUL_HTTP_SSL", ""); ssl != "" {
+			enabled, err := strconv.ParseBool(ssl)
+			if err != nil {
+				logFatal("consul ssl error", err)
+			}
+			if enabled {
+				options.TLS = &tls.Config{}
+			}
+		}
 	}
 	if url.Scheme == "etcd" {
 		etcd.Register()
 		sourceType = store.ETCD
 		client = env.Getenv("ETCD_ADDR", "localhost:2379")
+		if timeout := env.Getenv("ETCD_TIMEOUT", ""); timeout != "" {
+			num, err := strconv.ParseInt(timeout, 10, 16)
+			if err != nil {
+				logFatal("etcd timeout error", err)
+			}
+			options.ConnectionTimeout = time.Duration(num) * time.Second
+		}
+		if ssl := env.Getenv("ETCD_TLS", ""); ssl != "" {
+			enabled, err := strconv.ParseBool(ssl)
+			if err != nil {
+				logFatal("consul ssl error", err)
+			}
+			if enabled {
+				options.TLS = &tls.Config{}
+			}
+		}
+		options.Username = env.Getenv("ETCD_USERNAME", "")
+		options.Password = env.Getenv("ETCD_PASSWORD", "")
 	}
 	if url.Scheme == "zk" {
 		zookeeper.Register()
 		sourceType = store.ZK
 		client = env.Getenv("ZK_ADDR", "localhost:2181")
+		if timeout := env.Getenv("ZK_TIMEOUT", ""); timeout != "" {
+			num, err := strconv.ParseInt(timeout, 10, 16)
+			if err != nil {
+				logFatal("zk timeout error", err)
+			}
+			options.ConnectionTimeout = time.Duration(num) * time.Second
+		}
 	}
 	if url.Scheme == "boltdb" {
 		boltdb.Register()
 		sourceType = store.BOLTDB
 		client = url.Path
 		if client == "" {
-			client = env.Getenv("BOLTDB_PATH", "")
+			client = env.Getenv("BOLTDB_DATABASE", "")
```

Comment:
Ok, let's keep it simple and _start_ by only supporting `url.Path` - it's easy to add it later if necessary :)